src/create-zero.ts

```diff
@@ -0,0 +1,46 @@
+import type { CustomMutatorDefs, Query, Schema, ZeroOptions } from '@rocicorp/zero'
+import type { MaybeRefOrGetter, ShallowRef } from 'vue'
+import type { QueryResult, UseQueryOptions } from './query'
+import { Zero } from '@rocicorp/zero'
+import { shallowRef, toValue, watch } from 'vue'
+import { useQueryWithZero } from './query'
+
+export function createZero<S extends Schema = Schema, MD extends CustomMutatorDefs | undefined = undefined>(optsOrZero: MaybeRefOrGetter<ZeroOptions<S, MD> | { zero: Zero<S, MD> }>) {
```

Comment:
I think this should be something like: `createSyncedZeroComposables` (or a better name?) as semantically it does not create a zero instance.

Another approach would be, allowing passing a `zero` shallowRef as a first argument to `useQuery`, and then creating a 'bound' version like this:

```ts
const zero = createZero() // the same utility as `createZero` here, which lazily initialises zero on access
const useQuery = _useQuery.bind(zero)
// or
nuxtApp.provide('zero', zero)
function useQuery (opts) {
  const { $zero } = useNuxtApp()
  return _useQuery($zero, opts)
}
// or any number of other patterns
```